system/include/emscripten/wasmfs.h

```diff
@@ -14,6 +14,9 @@
 extern "C" {
 #endif
 
+#ifdef __cplusplus
+namespace wasmfs {
+#endif
```

Comment:
IIUC, as a result of this, C++ users need to write `wasmfs::wasmfs_mount` for example, which is redundant. Why do we need the namespace?

Reply:
The namespace issue came from trying to directly include `#include <emscripten/wasmfs.h>` in `js_api.cpp` instead of forward declaring methods like `wasmfs_create_directory`. Since `emscripten/wasmfs.h` also had to function as a normal C header, `backend_t` was redefined here. This meant using `backend_t` in `js_api.cpp` was ambiguous, since there was `backend_t` and `wasmfs::backend_t`. Based on https://github.com/emscripten-core/emscripten/pull/19825#discussion_r1275392176 I think it might be easier to use forward declarations for now and add a TODO.